apps/test-viewer/src/components/repositories-tree/RepositoriesService.ts

```diff
@@ -0,0 +1,72 @@
+/*---------------------------------------------------------------------------------------------
+ * Copyright (c) Bentley Systems, Incorporated. All rights reserved.
+ * See LICENSE.md in the project root for license terms and full copyright notice.
+ *--------------------------------------------------------------------------------------------*/
+
+import { IModelApp } from "@itwin/core-frontend";
+import { ITwinRepositoryType } from "./RepositoriesType";
+
+interface RepositoryData {
+  id?: string;
+  displayName: string;
+  type?: string;
+  name?: string;
+}
+
+interface ITwinRepository {
+  class: ITwinRepositoryType;
+  subClass?: string;
+  uri: string;
+}
+
+/**
+ * @internal
+ */
+export async function getItwinRepositories(itwinId: string, environment?: "PROD" | "QA" | "DEV"): Promise<ITwinRepository[]> {
+  const url = getRepositoriesUrl(itwinId, environment);
+  const result = (await fetchData(url)) as ITwinRepository[];
+  return result;
+}
+
+/**
+ * @internal
+ */
+export async function getRepositoryData(url: string): Promise<RepositoryData[]> {
+  const result = (await fetchData(url)) as RepositoryData[];
+  return result;
+}
+
+async function fetchData(url: string): Promise<unknown> {
+  const accessToken = await IModelApp.getAccessToken();
+  const headers = {
+    Authorization: accessToken,
+    "Content-Type": "application/json",
+    Prefer: "return=representation",
+  };
+
+  try {
+    const response = await fetch(url, {
+      headers,
+    });
+    const result = await response.json();
+
+    // get the data avoiding links
+    delete result._links;
+    const values = Object.values(result);
+    const data = values[0];
+
+    return data;
+  } catch (error) {
+    throw error;
+  }
+}
+
+function getRepositoriesUrl(itwinId: string, environment?: "PROD" | "QA" | "DEV") {
+  const prefix = {
+    QA: "qa-",
+    DEV: "dev-",
+    PROD: "",
+  };
+
+  return `https://${prefix[environment ?? "PROD"]}api.bentley.com/itwins/${itwinId}/repositories`;
```

Comment:
why not continue to use `globalThis.IMJS_URL_PREFIX` here?

Reply:
The idea was that this would be moved outside of test-viewer, with that intention would like not to use env directly.

Reply:
when we do move it out into a separate pkg, i still think we should avoid exposing our envs. we should just have a prop for apps to pass in a baseurl for these services

Reply:
Theoretically, if we hosted the service on some other endpoint or locally you should be able to point to that as well. I'm going to resolve this comment, but something we should def try to avoid when we publish this separately. 

We have been recommending users provide the url themselves if it's anything other than prod

Reply:
Didn't think about that, good point. Changed to use url instead of passing prefix.